vcr/credential/revocation.go

```diff
@@ -0,0 +1,77 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package credential
+
+import (
+	"time"
+
+	"github.com/nuts-foundation/go-did"
+)
+
+// Revocation defines a proof that a VC has been revoked by it's issuer.
+type Revocation struct {
+	// Issuer refers to the party that issued the credential
+	Issuer did.URI `json:"issuer"`
+	// Subject refers to the VC that is revoked
+	Subject did.URI `json:"subject"`
+	// CurrentStatus describes the current status, eg: 'Revoked'
+	CurrentStatus string `json:"currentStatus"`
```

Comment:
It was modelled after an existing revocation model (which I can't find anymore).
The others statuses would be `suspended`, `disputed` and `under investigation` but since we cannot support that (it would break the CRDT requirement) we could just use `reason` and `date`